include/util/integer_range.hpp

```diff
@@ -39,6 +39,7 @@ template <typename Integer> class range
 
     // Iterator functions
     bool operator!=(const range &) const noexcept { return iter < last; }
+    bool operator==(const range &other) const noexcept { return !operator!=(other); }
```

Comment:
Ticketed a long time ago: https://github.com/Project-OSRM/osrm-backend/issues/2859 this is biting us down the road..